lib/iex/lib/iex/evaluator.ex

```diff
@@ -254,6 +260,23 @@ defmodule IEx.Evaluator do
     :elixir_errors.parse_error(line, "iex", error, token)
   end
 
+  defp check_result(result, code) when is_list(result) do
+    if Enum.all?(result, &is_integer(&1)) do
```

Comment:
It is not all lists of integers but only list of integers within a certain range. Please check Inspect.List.